File: Data Structures & Algorithms/two-integer-sum-ii/submission-10.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        dictionary = {}
        for i, n in enumerate(numbers):
            diff = target - n
            if diff in dictionary:
                return [dictionary[diff] + 1, i + 1]
            else:
                dictionary[n] = i
        return []

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-10.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0, r = nums.size() - 1;
        while (l <= r) {
            int m = l + (r - l) / 2;
            cout << m << endl;
            if (nums.at(l) <= nums.at(m)) {
                if (nums.at(m) == target) {
                    return m;
                } else if (nums.at(l) <= target && target < nums.at(m)) {
                    r = m - 1;
                } else {
                    l = m + 1;
                }
            } else {
                if (nums.at(m) == target) {
                    return m;
                } else if (nums.at(m) < target && target <= nums.at(r)) {
                    l = m + 1;
                } else {
                    r = m - 1;
                }
            }
        }
        return -1;
    }
};
